src/client/components/Header/Header.tsx

```diff
@@ -1,14 +1,144 @@
 import React, { Component } from 'react';
+import './Header.css';
 
-export class Header extends Component {
-    constructor(props: any) {
-      super(props);
-      this.state = {};
+import {fetchProfileAPI} from '../../utils/API';
+import { getUsername } from '../../utils/auth';
+
+interface State {
+  dropdownHidden: boolean;
+  gravatarHash: string | undefined;
+  avatarOption: string | undefined;
+  gravatarDefault: string | undefined;
+  env: string | undefined;
+  envIcon: string | undefined;
+  username: string | undefined;
+  realname: string | undefined;
+}
+
+interface Props {
+  headerTitle: string;
+}
+
+export class Header extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      dropdownHidden: true,
+      gravatarHash: undefined,
+      avatarOption: undefined,
+      gravatarDefault: undefined,
+      env: undefined,
+      envIcon: undefined,
+      username: undefined,
+      realname: undefined,
+    };
+    this.getUserID = this.getUserID.bind(this);
+    this.setUrl_prefix = this.setUrl_prefix.bind(this);
+    this.getUserInfo = this.getUserInfo.bind(this);
+    this.dropDown = this.dropDown.bind(this);
+  }
+
+  componentDidMount(){
+    this.setUrl_prefix()
+    this.getUserID();
+    this.getUserInfo();
+  }
+
+  getUserID(){
+    getUsername((username: string) => {
+      window._env.username = username;
+    });
+  }
+
+  setUrl_prefix(){
+    let prefix: string;
+    let icon: string;
+    switch (window._env.url_prefix) {
+      case '':
+      case 'https:ci.kbase.us':
+        prefix = 'CI';
+        icon = 'fa fa-2x fa-flask';
+        break;
+      case 'https:appdev.kbase.us':
+        prefix = 'APPDEV';
+        icon = 'fa fa-2x fa-wrench';
+        break;
+      default:
+        prefix = 'CI';
     }
-  
-    render() {
-      return (
-        <div>I need this somewhere in header</div>
-      );
+    this.setState({env: prefix, envIcon: icon})
+  }
+
+  async getUserInfo(){
+    let res = await fetchProfileAPI();
```

Comment:
Unless you reassign the variable at some point, you should always use `const` instead of `let`.